### DISCUSSION TITLE
ISR Dynamic Page Serving Empty HTML Source

### DISCUSSION THREAD

**Discussion by st11x** [vercel/next.js#35921] [Help]
Hi,

I have a dynamic page /page/[[...path]].js and it uses getStaticProps with revalidate = 10 seconds. 

After a "npm run build", if I accessed say the page /news, the HTML source has empty elements like the HTML title field.

![image](https://user-images.githubusercontent.com/7398499/161850685-fe5790cf-f20e-416f-b985-cf7308ee0dfb.png)

I know there is a subsequent request to fetch the data and hydrate the page, but the initial page load returns the empty source.

The next page request after 10 seconds will have the correct title in the HTML source. If I ran "npm run build" again, then we are back to the first empty source page.

For pages that have not been visited first, my web crawler wouldn't find any data to index. 

How can I workaround this problem? 

Thanks
  Matt

**Comment by st11x** [comment]
I thought I could workaround the problem by running the web crawler twice, but requests made from the crawler does not update the cached page in NextJS. The request has to come from the browser.

**Comment by st11x** [comment]
This page https://nextjs.org/blog/next-12#bot-aware-isr-fallback suggests that NextJS is already aware and takes care of that for you. The problem was the user-agent coming from the web crawler and NextJS does not recognize that as a bot. 

Can this be configured in NextJS? I currently make the crawler identify as a Googlebot, not the best solution but it works.

**Comment by icyJoseph** [comment]
Don't you have to return `fallback: 'blocking'` in your `getStaticPaths` function? Do you have that already?

**Reply by st11x** [reply]
No, you shouldn't need to set fallback='blocking' if you are on NextJS 12. See the link I included.

The reason it is not working is because NextJS uses a hardcoded list of bots to test against. It will not know bots like Elasticsearch, Nutch, etc.

**Reply by icyJoseph** [reply]
Ah right, I misunderstood your use case, because there's a mistake on the link you've included, at least a diction mistake. I like the statement from the [docs](https://nextjs.org/docs/api-reference/data-fetching/get-static-paths#fallback-true) much better:

> The paths that have not been generated at build time will not result in a 404 page. Instead, Next.js will serve a [“fallback”](https://nextjs.org/docs/api-reference/data-fetching/get-static-paths#fallback-pages) version of the page on the first request to such a path. Web crawlers, such as Google, won't be served a fallback and instead the path will behave as in [fallback: 'blocking'](https://nextjs.org/docs/api-reference/data-fetching/get-static-paths#fallback-blocking) 

That being said, for your use case, like you say, unfortunately, the `isBot` function in the codebase does have a hard coded list, which makes your solution look like the best fit as of now.
